### DISCUSSION TITLE
Allow disabling certain auto complete suggestions.

### DISCUSSION THREAD

**Discussion by irvinsci** [forcedotcom/salesforcedx-vscode#5417] [Ideas]
**Is your feature request related to a problem? Please describe.**
The `@AuraEnabled` auto complete does not fit all use cases. When it is used in the wrong use case, I end up having to battle my muscle memory to not use the auto complete to populate incorrect code for my use case.

**Describe the solution you'd like**
Remove this auto complete suggestion. Or otherwise expose all auto complete suggestions in a settings page that users can customize.

**Describe alternatives you've considered**
I am tempted to disable this extension everytime this happens.

**Additional context**
### See that we use `@AuraEnabled` to describe variables
![Screenshot 2024-02-07 at 7 06 39 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/53797194/58c04097-0d02-4222-b538-91be68f92952)

### See how the `@AuraEnabled` auto complete always shows up in this case even when it is inappropriate
![Screenshot 2024-02-07 at 7 06 54 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/53797194/bed46aa4-b1d9-4dd6-a38a-a5c5925a7e01)

### See how any accidental key stroke can populate multiple unwanted lines require multiple keystrokes to cleanup
![Screenshot 2024-02-07 at 7 06 59 PM](https://github.com/forcedotcom/salesforcedx-vscode/assets/53797194/14700884-1445-4bfa-ae50-1df94f9ab011)

**Comment by mingxuanzhangsfdx** [comment]
Thank you @irvinsci for sending the feature request. We will let you know if we will implement the feature!

**Comment by AndrewStopchenko-SO** [comment]
Hi @irvinsci,

Those standard Apex snippets come from the `%user_profile%/.vscode/extensions/salesforce.salesforcedx-vscode-apex-##.##.#/snippets/apex.json` snippet file. The file is editable and you probably can remove/modify the standard snippet(s) there, but those changes will be lost after the extension gets updated to the newer version...

As a workaround, with the _VSCode: File > Preferences > Configure Snippets > New Global Snippets file..._, you can create an `apex.json` [user-defined snippet file](https://code.visualstudio.com/docs/editor/userdefinedsnippets) for Apex language, and add the following two snippets there (on Windows, I have this in the _%user_profile%/AppData/Roaming/Code/User/snippets/apex.json_ file):
```json
{
	"@AuraEnabled annotation": {
		"prefix": "@AuraEnabled",
		"body": [
			"@AuraEnabled"
		],
		"description": "@AuraEnabled annotation only"
	},

	"@AuraEnabled variable": {
		"prefix": "@AuraEnabled variable",
		"body": [
			"@AuraEnabled",
			"public ${1:varType} ${0:varName};"
		],
		"description": "Basic @AuraEnabled class variable"
	},
}
```

Unfortunately, it doesn't seem to allow overriding/shadowing the default Apex snippet for the `@AuraEnabled` prefix, but when we have a user-defined snippet that uses that same prefix text (or starts with that same prefix text), it will appear as a suggestion option - please mention the _(User snippet)_ in the suggestion preview's title.
![image](https://github.com/user-attachments/assets/e19a143e-0e54-43d2-937c-d035e30b342b)

The `@AuraEnabled variable` snippet is just for the "completeness" of the annotation options, allowing to insert:
- the **annotation only** (user snippet),
- the **annotated method** (standard snippet),
- the **annotated variable** (user snippet).
